src/cpp/src/llm_pipeline_static.cpp

```diff
@@ -179,6 +176,17 @@ StaticLLMPipeline::StaticLLMPipeline(
 ) : StaticLLMPipeline(path, path.string(), device, config) {
 }
 
+void StaticLLMPipeline::start_chat(const std::string& system_message) {
+    if (system_message.empty()) {
```

Comment:
Looks like here is a typo, system message should be added to history only if not empty.

```suggestion
    if (!system_message.empty()) {
```